config.go

```diff
@@ -0,0 +1,62 @@
+package timescaledb
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"go.k6.io/k6/lib/types"
+
+	"gopkg.in/guregu/null.v3"
+)
+
+type config struct {
+	// Connection URL in the form specified in the libpq docs,
+	// see https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING):
+	// postgresql://[user[:password]@][netloc][:port][,...][/dbname][?param1=value1&...]
+	URL          null.String
+	PushInterval types.NullDuration
+}
+
+func newConfig() config {
+	return config{
+		URL:          null.NewString("postgresql://localhost/myk6timescaleDB", false),
+		PushInterval: types.NewNullDuration(time.Second, false),
+	}
+}
+
+func (c *config) apply(cfg config) {
```

Comment:
@MStoykov is there any reason this config is not a pointer in most of the implementations? Like [here](https://github.com/grafana/k6/blob/master/output/influxdb/config.go#L68)

Reply:
yes, so it's always copied and you always return the new config that just the other was "applied", so please do it like that .

Reply:
Alright, I've changed it to return the copy.

(But I don't understand yet the reason that we are making copies when we could be manipulating the config directly.)

Reply:
I have apparently forgotten to answer this :sweat_smile: sorry.

Uh apart from this is what is done in general the [original place](https://github.com/grafana/k6/blob/16bc2b9267afbf7e555c59eb1857a10e53dc63e8/cmd/config.go#L185-L189) it was done had this problem where the first and the last Configuration that was applied was actually the same, see ow cliConf is the first thing that gets applied upon an the last thing to Apply upon. This is done as the defaults values are in the `cliConf` so we start with them and then we go through everything but they are also the thing with the highest priority ;). 
(for the record we would like to not do it like that and there is some work on it , you can look at issue 883 which I am not gonna link as it's quite huge already).

Additionally, this does help a lot in testing as it means that you can reuse some base config multiple times without fearing that a previous Apply would change it.

As for manipulating the config directly, while that will likely be slightly more performant this is an area where performance isn't at all interesting, so that benefit is in practice not interesting. 

There are probably other things that I have missed.